### PR TITLE
Add nested tenant attributes for registration

### PR DESCRIPTION
**Pull request by sviatoslav-krupa**

### CODE REVIEW

File: app/views/devise/shared/_registration_form.html.slim [hunk 1]
```diff
@@ -23,6 +23,8 @@
     .form-control-position
       i.fa.fa-envelope
     .help-block.font-small-3
+  = f.hidden_field :role, :value => User::ADMIN_ROLE
```
Comment (igorkasyanchuk) [inline app/views/devise/shared/_registration_form.html.slim]: move to controller. hidden fields for role is very bad practice. everyone could look in html body

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+describe User, type: :model do
```
Comment (igorkasyanchuk) [inline spec/models/user_spec.rb]: create Capybara spec
and test sign up - https://github.com/teamcapybara/capybara#using-capybara-with-rspec using it

File: app/models/user.rb [hunk 3]
```diff
@@ -1,37 +1,20 @@
...
 class User < ApplicationRecord
   SALE_ROLE = 'sale'.freeze
   ADMIN_ROLE = 'admin'.freeze
   MODERATOR_ROLE = 'moderator'.freeze
   ROLES = [SALE_ROLE, ADMIN_ROLE, MODERATOR_ROLE].freeze
+
   has_many :companies, dependent: :destroy
   belongs_to :tenant, optional: true
+
+  accepts_nested_attributes_for :tenant
```
Comment (igorkasyanchuk) [inline app/models/user.rb]: what happens if I'm sending from sing up page empty tenant name?
Admin must have tenant (add validation) for admin

File: app/controllers/users/registrations_controller.rb [hunk 4]
```diff
@@ -2,13 +2,25 @@ class Users::RegistrationsController < Devise::RegistrationsController
   before_action :configure_sign_up_params, only: [:create]
   before_action :configure_account_update_params, only: [:update]
 
+  def new
+    build_resource({})
+    resource.tenant = Tenant.new
```
Comment (Kmalynss) [inline app/controllers/users/registrations_controller.rb]: ```resource.build_tenant``` - please read API thoroughly https://apidock.com/rails/ActiveRecord/Associations/ClassMethods/belongs_to

File: app/controllers/users/registrations_controller.rb [hunk 5]
```diff
@@ -2,13 +2,25 @@ class Users::RegistrationsController < Devise::RegistrationsController
   before_action :configure_sign_up_params, only: [:create]
   before_action :configure_account_update_params, only: [:update]
 
+  def new
+    build_resource({})
```
Comment (Kmalynss) [inline app/controllers/users/registrations_controller.rb]: This attribute is optional. So no need to pass it.
Read API

**Pull request merged**